FirebaseAdmin/FirebaseAdmin.Tests/Messaging/MessageTest.cs

```diff
@@ -0,0 +1,237 @@
+            AssertJsonEquals(expected, message);
+        }
+
+        [Fact]
+        public void AndroidConfig()
+        {
+            var message = new Message()
+            {
+                Topic = "test-topic",
+                AndroidConfig = new AndroidConfig()
+                {
+                    CollapseKey = "collapse-key",
+                    Priority = Priority.High,
+                    TimeToLive = TimeSpan.FromMilliseconds(10),
+                    RestrictedPackageName = "test-pkg-name",
+                    Data = new Dictionary<string, string>()
+                    {
+                        { "k1", "v1" },
+                        { "k2", "v2" },
+                    },
+                },
+            };
+            var expected = new JObject()
+            {
+                {"topic", "test-topic"},
+                {
+                    "android", new JObject()
+                    {
+                        { "collapse_key", "collapse-key" },
+                        { "priority", "high" },
+                        { "ttl", "0.010000000s" },
+                        { "restricted_package_name", "test-pkg-name" },
+                        {"data", new JObject(){{"k1", "v1"}, {"k2", "v2"}}},
+                    }
+                },
+            };
+            AssertJsonEquals(expected, message);
+        }
+
+        [Fact]
+        public void AndroidConfigFullSecondsTTL()
+        {
+            var message = new Message()
+            {
+                Topic = "test-topic",
+                AndroidConfig = new AndroidConfig()
+                {
+                    TimeToLive = TimeSpan.FromHours(1),
+                },
+            };
+            var expected = new JObject()
+            {
+                {"topic", "test-topic"},
+                {
+                    "android", new JObject()
+                    {
+                        { "ttl", "3600s" },                        
+                    }
+                },
+            };
+            AssertJsonEquals(expected, message);
+        }
+
+        [Fact]
+        public void AndroidConfigInvalidTTL()
+        {
+            var message = new Message()
+            {
+                Topic = "test-topic",
+                AndroidConfig = new AndroidConfig()
+                {
+                    TimeToLive = TimeSpan.FromHours(-1),
+                },
+            };
+            var expected = new JObject()
+            {
+                {"topic", "test-topic"},
+                {
+                    "android", new JObject()
+                    {
+                        { "ttl", "3600s" },                        
+                    }
+                },
+            };
+            Assert.Throws<ArgumentException>(() => message.Validate());
+        }
+
+        private void AssertJsonEquals(JObject expected, Message actual)
+        {
+            var json = NewtonsoftJsonSerializer.Instance.Serialize(actual.Validate());
```

Comment:
You could use JsonConvert here as well.

Reply:
This is actually needed for serialization (ok for deserialization). By default `JsonConvert` serializes null/zero values, which we don't want in the output.

Reply:
You can use this btw:
```
string json = JsonConvert.SerializeObject(message, Formatting.None,
            new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore})```

Reply:
Thanks @DenSmoke. `NewtonsoftJsonSerializer` from Google API client basically handles this. This prevents us from having to write the settings boilerplate in our code.

https://github.com/googleapis/google-api-dotnet-client/blob/bb7e9a2af7684ccd07e09ce0b52c2d00794795bb/Src/Support/Google.Apis.Core/Json/NewtonsoftJsonSerializer.cs#L141-L147